### PR TITLE
Remove old kernel, bug fixes, and more!

### PR DESCRIPTION
**Pull request by AnErrupTion**
- Ported TestWorld as a BareBones kernel
- Removed Mosa.UnitTests.x86 and Mosa.Kernel.x86 projects
- Implemented Mosa.Kernel.BareMetal.BootSettings.EnableDebugOutput to fix unit testing
- Made the unit test engine not output anything to the invisible console
- Made the template use the BareMetal kernel
- Updated documentation to reflect all the changes
- Fixed BareMetal x86 unit test runner
- Added output message when running the disassemblers
- Moved projects under the Platforms solution folder into the Compiler solution folder
- Made BareMetal unit test runner use dynamic memory buffers

### CODE REVIEW

**Comment by charsleysa** [comment]
There's a bug somewhere, the unit tests are consistently timing out for optimization level 2, 3, and 4 on all platforms.

File: Source/Docs/demos.rst [hunk 1]
```diff
@@ -7,30 +7,21 @@ MOSA includes three demos applications to demostrate various capabilities of the
...
+.. image:: images/mosa-demo-helloworld.png
 
-SVGAWorld
+TestWorld
 =========
 
-The **SVGAWorld** demo uses the VMware SVGA II driver for graphics, and uses multiple other features (similar to the ones used by **CoolWorld**) offered by MOSA to make a nice demo.
-
-.. image:: images/mosa-demo-svgaworld.webp
+The **TestWorld** demo is a barebones demo meant to show how you can create a barebones kernel using MOSA.
```
Comment (tgiphil) [inline Source/Docs/demos.rst:27]: Let's discuss demo name. Also, barebones is minimalistic wording. Let's use our mission statement: "As operating systems share a lot of common groundwork, MOSA aims to standardize interfaces in order to foster portability and to provide both projects with basic implementations of these interfaces." to guide as.

File: Source/Mosa.BareBones.TestWorld.x86/Mosa.BareBones.TestWorld.x86.csproj [hunk 2]
```diff
@@ -10,23 +10,10 @@
...
 	<ItemGroup>
-		<ProjectReference Include="..\Mosa.Kernel.x86\Mosa.Kernel.x86.csproj" />
 		<ProjectReference Include="..\Mosa.Korlib\Mosa.Korlib.csproj" />
 		<ProjectReference Include="..\Mosa.Runtime.x86\Mosa.Runtime.x86.csproj" />
-		<ProjectReference Include="..\Mosa.Runtime\Mosa.Runtime.csproj" />
-		<ProjectReference Include="..\Mosa.UnitTests\Mosa.UnitTests.csproj" />
 	</ItemGroup>
-	<ItemGroup>
-		<Content Include="Asm\Return10.o">
-			<CopyToOutputDirectory>Always</CopyToOutputDirectory>
```
Comment (tgiphil) [inline Source/Mosa.BareBones.TestWorld.x86/Mosa.BareBones.TestWorld.x86.csproj:24]: Need this. It's our one test of this functionality.

File: Source/Mosa.Kernel.BareMetal/Serial.cs
Comment (tgiphil) [inline Source/Mosa.Kernel.BareMetal/Serial.cs:1]: Where did this go?

File: Source/Mosa.Linux.sln [hunk 3]
```diff
@@ -44,7 +38,7 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Mosa.Platform.x86", "Mosa.P
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Mosa.Utility.Disassembler", "Mosa.Utility.Disassembler\Mosa.Utility.Disassembler.csproj", "{E45FBA9E-6F57-4511-8977-D8045DE09622}"
 EndProject
-Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Mosa.Demo.TestWorld.x86", "Mosa.Demo.TestWorld.x86\Mosa.Demo.TestWorld.x86.csproj", "{577B8294-334B-4E02-BE1A-CBA8ABAF8939}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Mosa.BareBones.TestWorld.x86", "Mosa.BareBones.TestWorld.x86\Mosa.BareBones.TestWorld.x86.csproj", "{577B8294-334B-4E02-BE1A-CBA8ABAF8939}"
```
Comment (tgiphil) [inline Source/Mosa.Linux.sln:41]: DM

File: Source/Mosa.UnitTests.BareMetal.x86/UnitTestEngine.cs [hunk 4]
```diff
@@ -65,13 +60,10 @@ public static void Setup(ushort comPort)
...
 		QueueCurrent = Queue;
 
-		PendingCount = 0;
-		TestCount = 0;
-
 		QueueNext.Store32(0);
 	}
 
-	private static void SendByte(byte b) => Serial.Write(ComPort, b);
+	private static void SendByte(byte b) => Platform.Serial.Write(ComPort, b);
```
Comment (tgiphil) [inline Source/Mosa.UnitTests.BareMetal.x86/UnitTestEngine.cs:66]: Let's proxy this thru the Serial instead of directly to the Platform API.

**Review by tgiphil — CHANGES_REQUESTED** [review]

**Comment by AnErrupTion** [comment]
- Fixed Sparse Conditional Constant Propagation bug with sign/zero extensions (thanks @tgiphil!)

File: Source/Mosa.Compiler.Framework/Analysis/SparseConditionalConstantPropagation.cs [hunk 5]
```diff
@@ -953,6 +1005,62 @@ private static bool IntegerOperation(BaseInstruction instruction, ulong operand1
 		return false;
 	}
 
+	private static bool SignOrZeroExtend(BaseInstruction instruction, ulong operand1, out ulong result)
+	{
+		if (instruction == IRInstruction.SignExtend8x32)
+		{
+			result = (operand1 & 0x80) == 0 ? operand1 : operand1 | 0xFFFFFF00;
```
Comment (tgiphil) [inline Source/Mosa.Compiler.Framework/Analysis/SparseConditionalConstantPropagation.cs:1012]: Needs to be input value needs to be truncated first. 

Same issue with below.

**Comment by tgiphil** [comment]
Please re-base. Most changes should already be incorporated.

**Comment by tgiphil** [comment]
Given the number of file changes since this pull request, let's reopen this after it has been rebased.

**Pull request closed without merging**